nuclear/cmake/Modules/NUClearCompilerSettings.cmake

```diff
@@ -20,6 +20,5 @@ set(CMAKE_MACOSX_RPATH ON)
 set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_RPATH} lib/ ../lib/ bin/lib)
 
 if(NOT MSVC)
-  # Compilation must be done with c++14 for NUClear to work
-  add_compile_options(-std=c++14 -fPIC)
+  add_compile_options(-std=c++17 -fPIC)
```

Comment:
[CMAKE_CXX_STANDARD](https://cmake.org/cmake/help/v3.1/variable/CMAKE_CXX_STANDARD.html#variable:CMAKE_CXX_STANDARD)
[CMAKE_POSITION_INDEPENDENT_CODE](https://cmake.org/cmake/help/v3.0/variable/CMAKE_POSITION_INDEPENDENT_CODE.html#variable:CMAKE_POSITION_INDEPENDENT_CODE)

Reply:
Really our whole method of handling cxx flags and features needs to change, and this is the minimal change required to fix this. If you want to fix this properly it should be in another PR